internal/codespaces/api/api.go

```diff
@@ -1082,16 +1083,16 @@ func (a *API) setHeaders(req *http.Request) {
 
 // withRetry takes a generic function that sends an http request and retries
 // only when the returned response has a >=500 status code.
-func (a *API) withRetry(f func() (*http.Response, error)) (resp *http.Response, err error) {
-	for i := 0; i < 5; i++ {
-		resp, err = f()
+func (a *API) withRetry(f func() (*http.Response, error)) (*http.Response, error) {
+	bo := backoff.NewConstantBackOff(a.retryBackoff)
+	return backoff.RetryWithData(func() (*http.Response, error) {
+		resp, err := f()
 		if err != nil {
-			return nil, err
+			return nil, backoff.Permanent(err)
 		}
 		if resp.StatusCode < 500 {
-			break
+			return resp, backoff.Permanent(err)
```

Comment:
Just curious on the pattern here - why wouldn't this be `return resp, nil`?

Reply:
Good catch! That would be simpler. Will make that change.